python/run-tests.py

```diff
@@ -22,6 +22,10 @@
 from os import path
 
 
+# Update this when we upgrade the default to a new released version
+DEFAULT_SPARK = "4.0.1"
```

Comment:
can we extract this from sbt build directly?

Reply:
can I come back to this as a follow up? these changes are a major pain and I'd like to get this in without getting bogged down in the details

we already have other places where we've hardcoded stuff like this, maybe they can all be addressed at once

Reply:
tracking on https://github.com/delta-io/delta/issues/5326